src/vm/vm.rs

```diff
@@ -454,7 +455,7 @@ impl VM {
         self.done()
     }
 
-    pub fn ffi_call(&mut self) -> Result<(), Trace> {
+    pub unsafe fn ffi_call(&mut self) -> Result<(), Trace> {
```

Comment:
Oh, you made all those all unsafe I guess. 

I guess I just have to make two points to make / open-ended questions to consider:

1. The VM is low level, and will deal with low-level things like reading raw bytes as opcodes and so forth. This does not mean that those operations are unsafe; they may well be entirely well-defined within the context of the invariants and expectations we have set in place. For instance, if we statically verify bytecode beforehand, executing instructions is not unsafe.

2. Unsafe code is infectious, yes, but this does not mean that everything related to it is unsafe too. I'm well aware of the drama around actix web, and I think it's important to address that labeling unsafe code is as much as a social expectation as a technical one. With that said, I think there's a difference between compilers and web servers with respect to the social expectation of amount of unsafe code; and for this reason things that may be justifiably unsafe for one project may as well be regular business for another: I know the VM is low-level; I'm just not sure if that justifies marking the entire thing as unsafe. 

These two points are entirely subjective, of course 😄

On a slightly unrelated note, as I've mentioned earlier, the main reason most functions are public is for experimentation, access, and documentation. Once we figure out which items we want to export, we should make all other items private. Just putting this here so we don't forget later.